packages/wasm/crate/tests/test_metadata.rs

```diff
@@ -0,0 +1,164 @@
+        images: Vec::new(),
+        name: String::from(""),
+        penumbra_asset_id: None,
+        symbol: String::from(""),
+        priority_score: 0,
+    }
+}
+
+#[wasm_bindgen_test]
+fn it_interpolates_display_denom() {
+    assert_eq!(get_metadata_for("penumbra", false).base, "upenumbra");
+    assert_eq!(get_metadata_for("penumbra", false).display, "penumbra");
+    assert_eq!(
+        get_metadata_for("penumbra", false).denom_units[0].denom,
+        "upenumbra"
+    );
+    assert_eq!(
+        get_metadata_for("penumbra", false).denom_units[1].denom,
+        "penumbra"
+    );
+}
+
+use penumbra_wasm::metadata::customize_symbol_inner;
+
+#[wasm_bindgen_test]
+fn it_returns_non_staking_metadata_as_is() {
+    let metadata = Metadata {
+        name: String::from("Penumbra"),
+        symbol: String::from("UM"),
+        ..get_metadata_for("penumbra", false)
+    };
+    let customized_metadata = customize_symbol_inner(metadata.clone()).unwrap();
+
+    assert_eq!(metadata, customized_metadata);
+}
+
+#[wasm_bindgen_test]
+fn it_modifies_unbonding_token_symbol() {
+    let metadata = Metadata {
+        name: String::from("Unbonding Token"),
+        symbol: String::from(""),
+        ..get_metadata_for("unbonding_start_at_1234_penumbravalid1abcdef123456", false)
+    };
+    let customized_metadata = customize_symbol_inner(metadata.clone()).unwrap();
+
+    assert_eq!(customized_metadata.symbol, "unbondUMat1234(abcdef123456)");
+}
+
+#[wasm_bindgen_test]
+fn it_modifies_delegation_token_symbol() {
+    let metadata = Metadata {
+        name: String::from("Delegation Token"),
+        symbol: String::from(""),
+        ..get_metadata_for("delegation_penumbravalid1abcdef123456", false)
+    };
+    let customized_metadata = customize_symbol_inner(metadata.clone()).unwrap();
+
+    assert_eq!(customized_metadata.symbol, "delUM(abcdef123456)");
+}
+
+#[wasm_bindgen_test]
+fn it_modifies_auction_nft_symbol_with_seq_num() {
+    let metadata = Metadata {
+        name: String::from(""),
+        symbol: String::from(""),
+        ..get_metadata_for(
+            "auctionnft_0_pauctid1jqyupqnzznyfpq940mv0ac33pyx77s7af3kgdw4nstjmp3567dks8n5amh",
+            true,
+        )
+    };
+    let customized_metadata = customize_symbol_inner(metadata.clone()).unwrap();
+
+    assert_eq!(
+        customized_metadata.symbol,
+        "auction@0(jqyupqnzznyfpq940mv0ac33pyx77s7af3kgdw4nstjmp3567dks8n5amh)"
+    );
+
+    let metadata = Metadata {
+        name: String::from(""),
+        symbol: String::from(""),
+        ..get_metadata_for(
+            "auctionnft_123_pauctid1jqyupqnzznyfpq940mv0ac33pyx77s7af3kgdw4nstjmp3567dks8n5amh",
+            true,
+        )
+    };
+    let customized_metadata = customize_symbol_inner(metadata.clone()).unwrap();
+
+    assert_eq!(
+        customized_metadata.symbol,
+        "auction@123(jqyupqnzznyfpq940mv0ac33pyx77s7af3kgdw4nstjmp3567dks8n5amh)"
+    );
+}
+
+#[wasm_bindgen_test]
```

Comment:
todo: this one doesn't need the `#[wasm_bindgen_test]` attribute right? It can just be `#[test]`. Can you confirm with the other tests as well and if the containing functions don't use wasm bindgen, let's do the normal test attribute.

Reply:
I'm all for colocation, but in this case it's the straightforward way to run the tests with wasm-pack.

Unfortunately without the `#[wasm_bindgen_test]` attribute, the test won't run, so I have to leave it there.

Reply:
Oh wait a sec, think there is an issue with this resolution. We have two kinds of tests within the crate:
- Unit tests. Normal run-of-the-mill tests that run in a normal rust env via `cargo test`. Has the #[test] attribute.
- Headless browser wasm tests. These are run in a headless browser environment by wasm pack and necessary for functionality that specifically uses wasm_bindgen. Has the #[wasm_bindgen_test] attribute.

We should be able to run both of these tests. It will require two separate commands and it looks like our CI is only running the wasm-pack ones. It shouldn't be necessary to add #[wasm_bindgen_test] to every Rust test as this slows down local development.

Reply:
I see - I will revert this and make sure the CI runs both the unit and wasm-pack tests